test/integration/default/build/cloudtrail.tf

```diff
@@ -0,0 +1,230 @@
+resource "aws_s3_bucket" "trail_1_bucket" {
+  bucket        = "${terraform.env}-trail-01-bucket"
+  force_destroy = true
+
+  policy = <<POLICY
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "AWSCloudTrailAclCheck",
+            "Effect": "Allow",
+            "Principal": {
+              "Service": "cloudtrail.amazonaws.com"
+            },
+            "Action": "s3:GetBucketAcl",
+            "Resource": "arn:aws:s3:::${terraform.env}-trail-01-bucket"
+        },
+        {
+            "Sid": "AWSCloudTrailWrite",
+            "Effect": "Allow",
+            "Principal": {
+              "Service": "cloudtrail.amazonaws.com"
+            },
+            "Action": "s3:PutObject",
+            "Resource": "arn:aws:s3:::${terraform.env}-trail-01-bucket/*",
+            "Condition": {
+                "StringEquals": {
+                    "s3:x-amz-acl": "bucket-owner-full-control"
+                }
+            }
+        }
+    ]
+}
+POLICY
+}
+
+resource "aws_iam_role" "cloud_watch_logs_role" {
+  name = "${terraform.env}-cloud-watch-logs-role"
+
+  assume_role_policy = <<POLICY
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+          "Sid": "",
+          "Effect": "Allow",
+          "Principal": {
+            "Service": "cloudtrail.amazonaws.com"
+          },
+          "Action": "sts:AssumeRole"
+        }
+    ]
+}
+POLICY
+}
+
+resource "aws_iam_role_policy" "cloud_watch_logs_role_policy" {
+  depends_on = ["aws_iam_role.cloud_watch_logs_role"]
+
+  name = "${terraform.env}-cloud-watch-logs-role-policy"
+  role = "${terraform.env}-cloud-watch-logs-role"
+
+  policy = <<POLICY
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "AWSCloudTrailCreateLogStream",
+            "Effect": "Allow",
+            "Action": [
+                "logs:CreateLogStream"
+            ],
+            "Resource": [
+                "arn:aws:logs:us-east-1:${data.aws_caller_identity.creds.account_id}:log-group:${aws_cloudwatch_log_group.trail_1_log_group.name}:log-stream:${data.aws_caller_identity.creds.account_id}_CloudTrail_${data.aws_region.region.name}*"
```

Comment:
Here's the location triggering the IAM role permission issue - the `us-east-1` here needs to be parameterized, that is, replaced with `${data.aws_region.region.name}`.

I test in us-east-2, and this grants permission to work in us-east-1; thus the conflict.